### PR TITLE
Adding EOReader library

### PR DESCRIPTION
**Pull request by remi-braun**
Adding [EOReader](https://github.com/sertit/eoreader) library.
The documentation can be found [here](https://eoreader.readthedocs.io/en/latest).

## Introduction
**EOReader** is a remote-sensing opensource python library reading optical and SAR sensors, loading and stacking bands, clouds, DEM and spectral indices in a sensor-agnostic way. 

EOreader has been awarded by a [Success Story](https://earth.esa.int/eogateway/news/new-open-source-python-library-improves-rapid-mapping-services) on ESA's Earth Online website

A talks also has been held at GeoPython 2022, and here are the [slides](https://seafile.unistra.fr/f/be2b461af970465b903e/)

## Optical

It aims to allow the user to load in a pythonic-way optical bands, index and clouds.
You can see how EOReader works in real by looking at [this tutorial](https://eoreader.readthedocs.io/en/latest/notebooks/base.html).

|Constellations | Product Types | Use archive |
|--- | --- | --- |
|Sentinel-2 | L1C & L2A | ✅ |
|Sentinel-2 Theia | L2A | ✅ |
|Sentinel-3 SLSTR | RBT | ✅ |
|Sentinel-3 OLCI | EFR | ✅ |
|Landsat 8 OLCI | Level 1 | Collection 1: ❌, Collection 2: ✅ |
|Landsat 7 ETM | Level 1 | Collection 1: ❌, Collection 2: ✅ |
|Landsat 5 TM | Level 1 | Collection 1: ❌, Collection 2: ✅ |
|Landsat 4 TM  | Level 1 | Collection 1: ❌, Collection 2: ✅ |
|Landsat 5 MSS | Level 1 | Collection 1: ❌, Collection 2: ✅ |
|Landsat 4 MSS | Level 1 | Collection 1: ❌, Collection 2: ✅ |
|Landsat 3 MSS | Level 1 | Collection 1: ❌, Collection 2: ✅ |
|Landsat 2 MSS | Level 1 | Collection 1: ❌, Collection 2: ✅ |
|Landsat 1 MSS | Level 1 | Collection 1: ❌, Collection 2: ✅ |
|PlanetScope | L3A & L3B | ✅ |
|SkySat|  | (Collect Product) ortho_* | ✅ |
|Pleiades | SEN, PRJ, ORT & MOS | ✅ |
|SPOT 7 | SEN, PRJ, ORT & MOS | ✅ |
|SPOT 6 | SEN, PRJ, ORT & MOS | ✅ |
|Maxar  | Standard & Ortho | ✅ |

\* *Archived Landsat Collection-1 are not managed because of the tar.gz format, which is too slow to process. It is better to work on the extracted product.*

## SAR
It aims to allow the user to load in a pythonic-way orthorectified SAR GRD data, by binding ESA SNAP `gpt`.
You can see how EOReader works in real by looking at [this tutorial](https://eoreader.readthedocs.io/en/latest/notebooks/SAR.html).

| Constellations | Product Type | Handled |
| --- | --- | --- |
| `COSMO-Skymed` | SCS | ✅ |
| `COSMO-SkyMed` 1st Generation | DGM | ✅ |
| `COSMO-SkyMed` 2nd Generation | DGM | ⚠️ |
| `COSMO-SkyMed` | GEC, GTC | ⚠️ | 
| `ICEYE` | SLC | ✅ |
| `ICEYE` |GRD | ✅ | 
| `ICEYE` | ORTHO | 💤 |
| `RADARSAT Constellation Mission` | SLC | ⚠️ | 
| `RADARSAT Constellation Mission` | GRC, GCC, GCD | ⚠️ |
| `RADARSAT Constellation Mission` | GRD | ✅ | 
| `RADARSAT-2` | SLC | ✅| 
| `RADARSAT-2` | SGX, SCN, SCW,<br>SCF, SCS, SSG, SPG | ⚠️ |
| `RADARSAT-2` | SGF | ✅ |
| `Sentinel-1` | SLC | ✅ | 
| `Sentinel-1` | GRD | ✅ |
| `SAOCOM-1` | SLC | ✅ | 
| `SAOCOM-1` | ID | ⚠ |
| `SAOCOM-1` | GEC | ✅ |
| `SAOCOM-1` | GTC | ✅ |
| `TerraSAR-X`, `TanDEM-X`, `PAZ SAR` | SSC | ✅ | 
| `TerraSAR-X`, `TanDEM-X`, `PAZ SAR` | MGD | ✅ |
| `TerraSAR-X`, `TanDEM-X`, `PAZ SAR` | GEC | ⚠️ |
| `TerraSAR-X`, `TanDEM-X`, `PAZ SAR` | EEC | ✅ |

\**always given with a GRD image*

✅: Tested   
⚠️: Never tested, **use it at your own risk!**  
❌: Not handled   
💤: Waiting for the release  

## Context

This library has been created in the context of the Copernicus Emergency Management Service rapid mapping and risk and recovery.

In these activations, it is needed to deliver information (such as flood or fire delineations, landslides mapping, etc.) based on various sensors (optical and SAR). As every minute counts in production, it seemed crucial to harmonize the ground on which are built our production tools, in order to make them as sensor-agnostic as possible.

### CODE REVIEW

File: src/pygieons/ecosystem_connections.py [hunk 1]
```diff
@@ -292,4 +292,10 @@
     {"from": "xarray-spatial", "to": "datashader"},
     {"from": "xarray-spatial", "to": "xarray"},
     {"from": "xarray", "to": "numpy"},
+    {"from": "eoreader", "to": "rioxarray"},
+    {"from": "eoreader", "to": "xarray"},
+    {"from": "eoreader", "to": "rasterio"},
+    {"from": "eoreader", "to": "geopandas"},
+    {"from": "eoreader", "to": "pystac"},
+    {"from": "eoreader", "to": "dask"},
```
Comment (HTenkanen) [inline src/pygieons/ecosystem_connections.py]: Excellent, thanks for contributing! 🙂 Would you mind adding these connections in alphabetical order and move the "eoreader" lines earlier, i.e. after [line 33](https://github.com/pygieons/pygieons/blob/main/src/pygieons/ecosystem_connections.py#L33) now having ` {"from": "eomaps", "to": "pyepsg"}`. With this, the style (i.e. alphabetical order) would remain correct. 🙏🏻

File: src/pygieons/ecosystem_pkgs.py [hunk 2]
```diff
@@ -41,7 +41,7 @@
         "vector": ['geopy', 'h3', 'osmnet', 'osmnx', 'pyntcloud', 'pyrosm', 'snkit'],
         "raster": ['earthengine-api', 'easystac', 'eemont', 'lidar', 'odc-stac', 'planetary-computer', 'pymap3d',
                    'pyotb', 'pyrosar', 'pystac', 'pystac-client', 'radiant-mlhub', 'rio-hist', 'rio-mucho', 'rio-tiler',
-                   'salem', 'satpy', 'sentinelsat', 'stackstac', 'verde', 'xarray-sentinel', 'xyzservices'],
+                   'salem', 'satpy', 'sentinelsat', 'stackstac', 'verde', 'xarray-sentinel', 'xyzservices', 'eoreader'],
```
Comment (HTenkanen) [inline src/pygieons/ecosystem_pkgs.py]: Could you add 'eoreader' after `"eemont"`, so the list would be in alphabetical order? 🙏🏻

**Review by HTenkanen — CHANGES_REQUESTED** [review]
Super! Thanks for contributing and adding EOReader! 🙂 👍🏻 There are a couple of small changes needed to keep things nice and tidy 🙂

**Comment by remi-braun** [comment]
This should be OK now :)

**Review by HTenkanen — APPROVED** [review]
Looks great! Thanks for contributing! 🚀

**Pull request merged**